src/kudu/util/net/sockaddr.cc

```diff
@@ -71,30 +71,32 @@ bool Sockaddr::operator==(const Sockaddr& other) const {
 }
 
 bool Sockaddr::operator<(const Sockaddr &rhs) const {
-  return addr_.sin_addr.s_addr < rhs.addr_.sin_addr.s_addr;
+  return NetworkByteOrder::Load128(addr_.sin6_addr.s6_addr) <
+      NetworkByteOrder::Load128(rhs.addr_.sin6_addr.s6_addr);
 }
 
 uint32_t Sockaddr::HashCode() const {
-  uint32_t hash = Hash32NumWithSeed(addr_.sin_addr.s_addr, 0);
-  hash = Hash32NumWithSeed(addr_.sin_port, hash);
+  // Note: IPv6 addresses are 128 bits.
+  uint32_t hash = Hash32PointerWithSeed(addr_.sin6_addr.s6_addr, 0);
```

Comment:
What does Hash32PointerWithSeed do?
I am assuming it doesn't use the entire 128 bits.

Reply:
yea, I think you probably want Hash128to64 or just use one of the "string hash" functions and pass the sizeof

Reply:
I am using the 128bit to 64 as suggested.